api/service/v1alpha1/service.proto

```diff
@@ -10,17 +10,25 @@ service KargoService {
   rpc CreateStage(CreateStageRequest) returns (CreateStageResponse);
   rpc ListStages(ListStagesRequest) returns (ListStagesResponse);
   rpc GetStage(GetStageRequest) returns (GetStageResponse);
+  rpc UpdateStage(UpdateStageRequest) returns (UpdateStageResponse);
   rpc DeleteStage(DeleteStageRequest) returns (DeleteStageResponse);
   rpc PromoteStage(PromoteStageRequest) returns (PromoteStageResponse);
   rpc ListProjects(ListProjectsRequest) returns (ListProjectsResponse);
 }
 
-message CreateStageRequest {
+message TypedStageSpec {
```

Comment:
Just checking if this is what the UI is expected to send if yes then we need separate fields for adding labels/annotations because that will be needed bu users.